### ISSUE TITLE
Visualscript connection lines get drawn over or below the nodes each time a node gets selected

### ISSUE THREAD

**Issue by mrcdk** [godotengine/godot#11639]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**

**Issue description:**

![https://i.imgur.com/zJhmmrO.gifv](https://i.imgur.com/zJhmmrO.gif)

**Steps to reproduce:**
1. Add nodes to a visualscript
2. Make lines overlap nodes to better see the issue
3. Start selecting different nodes (or the same sometimes)
4. Lines get drawn over and below the overlapping node each time you select a node

**Comment by akien-mga** [comment]
We have now entered release freeze for Godot 3.0 and want to focus only on release critical issues for that milestone. Therefore, we're moving this issue to the 3.1 milestone, though a fix may be made available for a 3.0.x maintenance release after it has been tested in the master branch during 3.1 development. If you consider that this issue is critical enough to warrant blocking the 3.0 release until fixed, please comment so that we can assess it more in-depth.

**Comment by akien-mga** [comment]
Is this still reproducible in the current master branch?

**Comment by mrcdk** [comment]
Yes, still happening in https://github.com/godotengine/godot/commit/382f0991ee7ece2a99cfbdbf641d46777b8c8edf

![https://i.imgur.com/GlUy5b2.gifv](https://i.imgur.com/GlUy5b2.gif)

**Comment by akien-mga** [comment]
I tried a bit in 3.3-stable and didn't manage to reproduce the issue, could you confirm if it's also fixed on your end?

I've been trying with the setup and all kinds of attempts at selecting different or the same nodes:

![Screenshot_20210423_223717](https://user-images.githubusercontent.com/4701338/115927271-775aa680-a484-11eb-9781-fefd2692262d.png)

**Comment by mrcdk** [comment]
Sadly it seems it's still happening on 3.3-stable.

![NN71UuaU4q](https://user-images.githubusercontent.com/483531/115929593-5bf19a80-a488-11eb-9771-360f1332c096.gif)

The gif was recorded on Windows 10 GLES3

**Comment by akien-mga** [comment]
Now I can also reproduce it... not sure what I missed before.

**Comment by Zylann** [comment]
I have seen this happen in my own tools, this is a problem with GraphEdit
